sdk/protocol2.go

```diff
@@ -0,0 +1,173 @@
+package sdk
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/newrelic/infra-integrations-sdk/args"
+	"github.com/newrelic/infra-integrations-sdk/cache"
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/newrelic/infra-integrations-sdk/metric"
+)
+
+// Entity is the producer of the data. Entity could be a host, a container, a pod, or whatever unit of meaning.
+type Entity struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
+}
+
+// EntityData defines all the data related to a particular event from an entity.
+type EntityData struct {
+	Entity    Entity             `json:"entity"`
+	Metrics   []metric.MetricSet `json:"metrics"`
+	Inventory Inventory          `json:"inventory"`
+	Events    []Event            `json:"events"`
+}
+
+// EntitiesData is a collection of EntityData.
+type EntitiesData []*EntityData
+
+// Has tells you if that Entity is already part of the data.
+func (d EntitiesData) Has(entity Entity) bool {
+	for _, e := range d {
+		if e.Entity.Name == entity.Name && e.Entity.Type == entity.Type {
```

Comment:
shouldn't we check that's empty also?

Reply:
We could but would not be responsibility of this method but `Entity` one.